src/names.c

```diff
@@ -0,0 +1,89 @@
+#include <glib.h>
+
+#include "names.h"
+#include "parse.h"
+
+/* Non-static as we need it for ABI compatibility, see at the end of the file */
+const char* const np_backend_to_str[NETPLAN_BACKEND_MAX_] = {
+        [NETPLAN_BACKEND_NONE] = "none",
+        [NETPLAN_BACKEND_NETWORKD] = "networkd",
+        [NETPLAN_BACKEND_NM] = "NetworkManager",
+        [NETPLAN_BACKEND_OVS] = "OpenVSwitch",
+};
+
+static const char* const np_wifi_mode_to_str[NETPLAN_WIFI_MODE_MAX_] = {
```

Comment:
```suggestion
static const char* const
netplan1_wifi_mode_to_str[NETPLAN_WIFI_MODE_MAX_] = {
```

Hmm... I don't really like the `np_*` API naming too much, as I think it is too generic. How about using a `netplan1_*` prefix? As we are on libnetplan v0 right now. And this could be the first functions of our libnetplan v1 API. It would even allow to keep backwards compatibility using `libnetplan1_X` and `libnetplan2_X` if we want/need to break API in the future.
What do you think?

Also, starting the symbol name in a new line is a good practice/code-style IMO. We're not fully consistent with that in the current codebase, but let's try to stick to that convention for new definitions. (That applies to all functions in this file)

Reply:
ACK for the symbol name at the start of the line.

After a quick verbal exchange with @slyon and @vorlonofportland we'll stick with `netplan_` as the prefix of our public API, as it is the more elegant solution from the user's PoV, even if it makes my life a bit harder ;-). I'll update this PR (and all the others...) shortly.